sdk/servicebus/azure-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,424 @@
+
+## Troubleshooting authentication issues
+
+### Authentication errors
+
+Authentication errors typically occur when the credentials provided are incorrect or have expired.
+
+**Common causes:**
+- Incorrect connection string
+- Expired SAS token
+- Invalid managed identity configuration
+- Wrong credential type being used
+
+**Resolution:**
+1. Verify your connection string is correct and complete
+2. Check if using SAS tokens that they haven't expired
+3. For managed identity, ensure the identity is properly configured and has the necessary permissions
+4. Test connectivity using a simple connection string first
+
+```python
+# Example of proper authentication
+from azure.servicebus import ServiceBusClient
+
+# Using connection string
+client = ServiceBusClient.from_connection_string("your_connection_string")
+
+# Using Azure Identity
+from azure.identity import DefaultAzureCredential
+credential = DefaultAzureCredential()
+client = ServiceBusClient("your_namespace.servicebus.windows.net", credential)
+```
+
+### Authorization errors
+
+Authorization errors occur when the authenticated identity doesn't have sufficient permissions.
+
+**Required permissions for Service Bus operations:**
+- **Send:** Required to send messages to queues/topics
+- **Listen:** Required to receive messages from queues/subscriptions  
+- **Manage:** Required for management operations (create/delete entities)
+
+**Resolution:**
+1. Check the Access Control (IAM) settings in Azure portal
+2. Ensure the identity has the appropriate Service Bus roles:
+   - `Azure Service Bus Data Owner`
+   - `Azure Service Bus Data Sender`
+   - `Azure Service Bus Data Receiver`
+3. For connection strings, verify the SAS policy has the correct permissions
+
+### Connection string issues
+
+**Common connection string problems:**
+- Missing required components (Endpoint, SharedAccessKeyName, SharedAccessKey)
+- Incorrect namespace or entity names
+- URL encoding issues with special characters
+
+**Example of correct connection string format:**
+```
+Endpoint=sb://your-namespace.servicebus.windows.net/;SharedAccessKeyName=your-policy;SharedAccessKey=your-key
+```
+
+## Troubleshooting connectivity issues
+
+### Connection errors
+
+Connection errors can occur due to network issues, firewall restrictions, or service problems.
+
+**Common causes:**
+- Network connectivity issues
+- DNS resolution problems
+- Firewall or proxy blocking connections
+- Service Bus namespace not accessible from current location
+
+**Resolution:**
+1. Test basic network connectivity to `your-namespace.servicebus.windows.net` on port 5671 (AMQP) or 443 (AMQP over WebSockets)
+2. Try using AMQP over WebSockets if regular AMQP is blocked:
+
+```python
+from azure.servicebus import ServiceBusClient, TransportType
+
+client = ServiceBusClient.from_connection_string(
```

Comment:
should use conn string

Reply:
This should use Identity auth and not conn str

Reply:
This should use Identity auth and not conn str